trie/database.go

```diff
@@ -355,22 +357,19 @@ func (db *Database) Get(key []byte) ([]byte, error) {
 	return zdb.Get(key)
 }
 
-func (db *Database) IsZk() bool {
-	if db.config == nil {
-		return false
-	}
-	return db.config.Zktrie
-}
+func (db *Database) IsZk() bool     { return db.config.Zktrie }
+func (db *Database) IsZkTree() bool { return db.config.Zktrie && db.config.ExperimentalZkTree }
```

Comment:
Why both `Zktrie` and `ExperimentalZkTree` should be true for `IsZkTree` being true?
I think these can be separated and one can be overwhelm the other.

Reply:
`Zktrie` is the value set from genesis.
`ExperimentalZkTree` is a geth execution parameter, so the condition that both must be enabled seemed reasonable.